executor/analyze.go

```diff
@@ -235,23 +269,21 @@ func warnLockedTableMsg(sessionVars *variable.SessionVars, needAnalyzeTableCnt u
 	}
 }
 
-func getTableIDFromTask(task *analyzeTask) int64 {
-	var tableID statistics.AnalyzeTableID
-
+func getTableIDFromTask(task *analyzeTask) statistics.AnalyzeTableID {
 	switch task.taskType {
 	case colTask:
-		tableID = task.colExec.tableID
+		return task.colExec.tableID
 	case idxTask:
-		tableID = task.idxExec.tableID
+		return task.idxExec.tableID
 	case fastTask:
-		tableID = task.fastExec.tableID
+		return task.fastExec.tableID
 	case pkIncrementalTask:
-		tableID = task.colIncrementalExec.tableID
+		return task.colIncrementalExec.tableID
 	case idxIncrementalTask:
-		tableID = task.idxIncrementalExec.tableID
+		return task.idxIncrementalExec.tableID
 	}
 
-	return tableID.TableID
+	panic("unreachable")
```

Comment:
Better to remove the panic, it may cause the process to exit if we add some new task but forget to update here.

Reply:
Do you have any suggestions on how to handle this case?